src/lib.rs

```diff
@@ -1,6 +1,560 @@
+        va.iter()
+        .zip(vb)
+        .all(|(a,b)| eq_with_nan_eq(a,b))
+    }
+
+
+    use crate::{StringDiffAlgorithm, StringDiffOp, HammingDistance};
+
+    #[test]
+    fn test_LevenshteinDistance_edit_distance(){
+        let test_struct = super::LevenshteinDistance{};
+
+        assert_eq!(3 , test_struct.distance("sets", "reset"));
+        assert_eq!(3 , test_struct.distance("sitting", "kitten"));
+        assert_eq!(3 , test_struct.distance("Sunday", "Saturday"));
+    }
+
+    #[test]
+    fn test_LevenshteinDistance_opp_distance(){
+        let test_struct = super::LevenshteinDistance{};
+
+        let mut test_vec: Vec<StringDiffOp> = Vec::new();
+        test_vec.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Insert('g'),
+            index: -1
+        });
+        test_vec.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Substitute('e','i'),
+            index: 4
+        });
+        test_vec.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Substitute('k','s'),
+            index: 0
+        });
+
+        let mut test_vec_2: Vec<StringDiffOp> = Vec::new();
+        test_vec_2.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Substitute('r','n'),
+            index: 4
+        });
+        test_vec_2.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Delete('t'),
+            index: 2
+        });
+        test_vec_2.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Delete('a'),
+            index: 1
+        });
+
+        let mut test_vec_3: Vec<StringDiffOp> = Vec::new();
+        test_vec_3.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Insert('S'),
+            index: -1
+        });
+        test_vec_3.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Delete('E'),
+            index: 1
+        });
+        test_vec_3.push(StringDiffOp {
+            kind: super::StringDiffOpKind::Delete('R'),
+            index: 0
+        });
+
+        assert_eq!(vec_compare(&test_vec, &test_struct.diff("sitting", "kitten")) , true );
+        assert_eq!(vec_compare(&test_vec_2, &test_struct.diff("Sunday", "Saturday")) , true );
+        assert_eq!(vec_compare(&test_vec_3, &test_struct.diff("SETS", "RESET")) , true );
+    }
+
+    #[test]
+    fn test_HammingDistance_edit_distance(){
+        let test_struct = super::HammingDistance{};
+
+        assert_eq!(3 , test_struct.distance("karolin", "kathrin"));
+        assert_eq!(3 , test_struct.distance("karolin", "kerstin"));
+        assert_eq!(4 , test_struct.distance("kathrin", "kerstin"));
+        assert_eq!(4 , test_struct.distance("0000", "1111"));
+        assert_eq!(3 , test_struct.distance("2173896", "2233796"));
+    }
+
+    #[test] 
+    fn test_HammingDistance_opp_distance(){
+        let test_struct = super::HammingDistance{};
+
+        let mut test_vec: Vec<StringDiffOp> = Vec::new();
+        test_vec.push(
+                super::StringDiffOp{
```

Comment:
These calls can be made more succinct after creating the utility constructors, e.g this could be `StringDiffOp::new_substitute()`